sources/Metal/Renderers/iTermCursorGuideRenderer.m

```diff
@@ -106,19 +126,30 @@ - (void)setColor:(NSColor *)color {
 - (void)drawWithFrameData:(iTermMetalFrameData *)frameData
            transientState:(__kindof iTermMetalCellRendererTransientState *)transientState {
     iTermCursorGuideRendererTransientState *tState = transientState;
-    if (tState.row < 0) {
-        // Cursor is offscreen. We set it to -1 to signal this.
+    if (tState.row < 0 || tState.col < 0) {
         return;
     }
 
     [tState initializeVerticesWithPool:_cellRenderer.verticesPool];
-    [_cellRenderer drawWithTransientState:tState
-                            renderEncoder:frameData.renderEncoder
-                         numberOfVertices:6
-                             numberOfPIUs:0
-                            vertexBuffers:@{ @(iTermVertexInputIndexVertices): tState.vertexBuffer }
-                          fragmentBuffers:@{}
-                                 textures:@{ @(iTermTextureIndexPrimary): tState.texture } ];
+    if (tState.row >= 0 && self.enabled) {
+        [_cellRenderer drawWithTransientState:tState
+                                renderEncoder:frameData.renderEncoder
+                             numberOfVertices:6
+                                 numberOfPIUs:0
+                                vertexBuffers:@{ @(iTermVertexInputIndexVertices): tState.vertexBuffer }
+                              fragmentBuffers:@{}
+                                     textures:@{ @(iTermTextureIndexPrimary): tState.texture } ];
+    }
+
+    if (tState.col >= 0 && self.venabled) {
+        [_cellRenderer drawWithTransientState:tState
+                                renderEncoder:frameData.renderEncoder
+                             numberOfVertices:6
+                                 numberOfPIUs:0
+                                vertexBuffers:@{ @(iTermVertexInputIndexVertices): tState.vvertexBuffer }
+                              fragmentBuffers:@{}
+                                     textures:@{ @(iTermTextureIndexPrimary): tState.texture } ];
```

Comment:
You need to use a different texture for the vertical guide so it will look right. In `initializeTransientState:`, the `_texture` is created using `newCursorGuideTextureWithTransientState:`. It draws a background and then two lines on the top and bottom. You'll need to modify it to draw the texture rotated 90° for the vertical guide's texture. As it stands, the horizontal texture gets stretched out the whole height of the view and it looks funny.

Reply:
Done.